flow_screen_components/quickChoiceFSCproject/force-app/main/default/lwc/quickChoiceFSC/quickChoiceFSC.js

```diff
@@ -1,167 +1,215 @@
+			console.log("setting Picklist on");
+			this.showRadio = false;
+		}
+
+		console.log("initializing smartChoice. inputMode is: " + this.inputMode);
+		let options = [];
+		let items = [];	//parameters for visual text box selection
+		let index = 0;
+		if (this.legitInputModes.includes(this.inputMode)) {
+			switch (this.inputMode) {
+				//User can simply pass in a collection of strings as choiceValues. The same text is used for both label and value
+				case "Single String Collection":
+					console.log("entering input mode String Collection");
+					console.log("choiceValues is: " + this.choiceValues);
+					//console.log ('splitting choice values would be: ' + this.choiceValues.split(','));
+					//let values = this.choiceValues.split(';');
+
+					this.choiceValues.forEach(value => {
+						console.log("value is: " + value);
+						options.push({ label: value, value: value });
+						console.log("options is: " + options);
+					});
+					break;
+
+				//User can  pass in one collection of strings for visible labels and another for the underlying values (such as recordIds)
+				case "Dual String Collections":
+					console.log("entering input mode Dual String Collections");
+					console.log("choiceValues is: " + this.choiceValues);
+					//console.log ('splitting choice values would be: ' + this.choiceValues.split(','));
+					//let values = this.choiceValues.split(';');
+
+					this.choiceLabels.forEach(label => {
+						console.log("label is: " + label);
+						console.log("value is: " + this.choiceValues[index]);
+						options.push({ label: label, value: this.choiceValues[index] });
+						console.log("options is: " + options);
+						index += 1;
+					});
+					break;
+				
+				//User passes in Label collection of string for box header and Value collection of strings for box description
+				case "Visual Text Box":
+					console.log("entering input mode Visual Text Box");
+					console.log("choiceLabels is: " + this.choiceLabels);
+					this.choiceLabels.forEach(label => {
+						items.push({ name: label, description: this.choiceValues[index], icon: this.choiceIcons[index] });
+						console.log("items is: " + items);
+						index +=1;
+					});
+					break;
+
+				default:
+			}
+			this.options = options;
+			this.items = items;
+		} else {
+			console.log(
+				"SmartChoiceFSC: Need a valid Input Mode value. Didn't get one"
+			);
+			throw new Error(
+				"SmartChoiceFSC: Need a valid Input Mode value. Didn't get one"
+			);
+		}
+	}
+
+	@api
+	validate() {
+		//If the component is invalid, return the isValid parameter as false and return an error message.
+		console.log("entering validate: required=" + this.required + " value=" + this.value);
+		let errorMessage = "You must make a selection in: " + this.masterLabel + " to continue";
+
+		if (this.required === true && !this.value) {
+			return {
+				isValid: false,
+				errorMessage: errorMessage
+			};
+		}
+
+		return { isValid: true };
+	}
+
+	handleChange(event) {
```

Comment:
notice how handleChange and handleRadio are nearly duplicates? pro developers develop a spidey sense for things like that and treat them as opportunities to combine functionality to DRY things up. in this case, can you combine them into a single handleChange that sets the selectedValue appropriately based on something in the event? (or is event.detail.value definitely different from event.target value? )

Reply:
They are different for input type=radio vs the lightning-combobox or radio-group.  I removed handleRadio and changed the assignment in handleChange to this:
this.selectedValue = (this.showVisual) ? event.target.value : event.detail.value;

Reply:
perfect resolution. cleaner, shorter code.